File: claude/nivel intermedio/ruby/numero.rb
# Función que calcula el promedio de una lista de números.
#
# @param numeros [Array<Float>] Lista de números
# @return [Float] Promedio de los números o 0 si la lista está vacía
def calcular_promedio(numeros)
  return 0 if numeros.empty?
  
  suma = numeros.sum
  return suma.to_f / numeros.length
end

# Solicitar valores al usuario
print "Ingrese una lista de números separados por comas: "
entrada = gets.chomp.strip

# Procesar la entrada del usuario
numeros = []
if !entrada.empty?
  numeros = entrada.split(',').map do |valor|
    begin
      valor.strip.to_f
    rescue
      puts "Error: valor inválido ignorado: #{valor}"
      nil
    end
  end.compact
end

# Calcular y mostrar el resultado
resultado = calcular_promedio(numeros)
puts "El promedio de la lista #{numeros.inspect} es: #{resultado}"

File: claude/nivel intermedio/python/numero.py
def calcular_promedio(numeros):
    """
    Función que calcula el promedio de una lista de números.
    
    Args:
        numeros (list): Lista de números
        
    Returns:
        float: Promedio de los números o 0 si la lista está vacía
    """
    if not numeros:
        return 0
    
    return sum(numeros) / len(numeros)

# Solicitar valores al usuario
entrada = input("Ingrese una lista de números separados por comas: ")

# Procesar la entrada del usuario
if entrada.strip():
    numeros = [float(x.strip()) for x in entrada.split(",")]
else:
    numeros = []

# Calcular y mostrar el resultado
resultado = calcular_promedio(numeros)
print(f"El promedio de la lista {numeros} es: {resultado}")